src/main/java/com/minecraftabnormals/abnormals_core/core/events/listener/ACEvents.java

```diff
@@ -59,4 +73,26 @@ public static void onUpdateEntity(LivingEvent.LivingUpdateEvent event) {
 			if (entity.getPersistentData().getBoolean(POISON_TAG)) ((IAgeableEntity) entity).resetGrowthProgress();
 		}
 	}
+
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public static void onNoteBlockPlay(NoteBlockEvent.Play event) {
+		BlockPos pos = event.getPos();
+		BlockState state = event.getWorld().getBlockState(pos.relative(Direction.DOWN));
+		for (DataUtil.CustomNoteBlockInstrument instrument : SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS) {
+			if (instrument.test(state)) {
+				SoundEvent sound = instrument.sound();
+				int note = event.getVanillaNoteId();
```

Comment:
`note` is only ever used after getting converted to a `double`, meaning an initial casting of `event.getVanillaNoteId()` to a `double` would be better.

Reply:
Done.